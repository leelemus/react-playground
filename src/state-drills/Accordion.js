import React from 'react';

class Accordion extends React.Component {

    static defaultProps = {sections: []};

    state = {
        currentIndex: null,
    };

    renderButtonSection(section, index, currentIndex) {
        return (
            <li key={index}>
                <button onClick = {() => this.buttonHandler(index)}>
                    {section.title}
                </button>
                {(currentIndex === index) && <p>{section.content}</p>}
            </li>
        )
    }

    buttonHandler (index) {
        this.setState({currentIndex: index});
    }

    render() {
        const {currentIndex} = this.state;
        const { sections } = this.props;

        return (
            <ul>
                {sections.map((section, index) => 
                    this.renderButtonSection(section, index, currentIndex)
                )}
            </ul>
        );
    }

}

export default Accordion;