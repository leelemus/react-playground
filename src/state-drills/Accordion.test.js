import React from 'react';
import ReactDOM from 'react-dom';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import Accordion from "./Accordion";

const sections = [
    {
      title: 'Section 1',
      content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
    },
    {
      title: 'Section 2',
      content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
    },
    {
      title: 'Section 3',
      content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
    },
];

describe ('Accordion Component', () => {
    it('renders without error', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Accordion />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('render no section by default', () => {
        const wrapper = shallow(<Accordion sections={sections} />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('opens correct section when clicked', () => {
        const wrapper = shallow(<Accordion sections={sections} />);
        wrapper.find('button').at(1).simulate('click');
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('only opens one sections at a time', () => {
        const wrapper = shallow(<Accordion section={sections} />);
        wrapper.find('p').at(1).simulate('click');
        wrapper.find('p').at(2).simulate('click');
        expect(toJson(wrapper)).toMatchSnapshot();
    });
})