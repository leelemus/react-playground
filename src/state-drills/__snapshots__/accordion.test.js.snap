// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion Component opens correct section when clicked 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
    <p>
      Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!
    </p>
  </li>
  <li
    key="2"
  >
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;

exports[`Accordion Component render no section by default 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li
    key="2"
  >
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;
